cmd/gctcli/commands.go

```diff
@@ -2357,3 +2401,407 @@ func disableExchangePair(c *cli.Context) error {
+	}
+
+	assetType = strings.ToLower(assetType)
+
+	if !validAsset(assetType) {
+		return errInvalidAsset
+	}
+
+	conn, err := setupClient()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	p := currency.NewPairDelimiter(pair, pairDelimiter)
+
+	client := gctrpc.NewGoCryptoTraderClient(conn)
+	result, err := client.GetOrderbookStream(context.Background(),
+		&gctrpc.GetOrderbookStreamRequest{
+			Exchange: exchangeName,
+			Pair: &gctrpc.CurrencyPair{
+				Base:      p.Base.String(),
+				Quote:     p.Quote.String(),
+				Delimiter: p.Delimiter,
+			},
+			AssetType: assetType,
+		},
+	)
+
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := result.Recv()
+		if err != nil {
+			return err
+		}
+
+		err = clearScreen()
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Orderbook stream for %s %s:\n\n", exchangeName,
+			resp.Pair.String())
+		fmt.Println("\t\tBids\t\t\t\tAsks")
+		fmt.Println()
+
+		bidLen := len(resp.Bids) - 1
+		askLen := len(resp.Asks) - 1
+
+		var maxLen int
+		if bidLen >= askLen {
+			maxLen = bidLen
+		} else {
+			maxLen = askLen
+		}
+
+		for i := 0; i < maxLen; i++ {
+			var bidAmount, bidPrice float64
+			if i <= bidLen {
+				bidAmount = resp.Bids[i].Amount
+				bidPrice = resp.Bids[i].Price
+			}
+
+			var askAmount, askPrice float64
+			if i <= askLen {
+				askAmount = resp.Asks[i].Amount
+				askPrice = resp.Asks[i].Price
+			}
+
+			fmt.Printf("%f %s @ %f %s\t\t%f %s @ %f %s\n",
+				bidAmount,
+				resp.Pair.Base,
+				bidPrice,
+				resp.Pair.Quote,
+				askAmount,
+				resp.Pair.Base,
+				askPrice,
+				resp.Pair.Quote)
+
+			if i >= 49 {
+				// limits orderbook display output
+				break
+			}
+		}
+	}
+}
+
+var getExchangeOrderbookStreamCommand = cli.Command{
+	Name:      "getexchangeorderbookstream",
+	Usage:     "gets a stream for all orderbooks associated with an exchange",
+	ArgsUsage: "<exchange>",
+	Action:    getExchangeOrderbookStream,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "exchange",
+			Usage: "the exchange to get the orderbook from",
+		},
+	},
+}
+
+func getExchangeOrderbookStream(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		cli.ShowCommandHelp(c, "getexchangeorderbookstream")
+		return nil
+	}
+
+	var exchangeName string
+	if c.IsSet("exchange") {
+		exchangeName = c.String("exchange")
+	} else {
+		exchangeName = c.Args().First()
+	}
+
+	if !validExchange(exchangeName) {
+		return errInvalidExchange
+	}
+
+	conn, err := setupClient()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	client := gctrpc.NewGoCryptoTraderClient(conn)
+	result, err := client.GetExchangeOrderbookStream(context.Background(),
+		&gctrpc.GetExchangeOrderbookStreamRequest{
+			Exchange: exchangeName,
+		})
+
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := result.Recv()
+		if err != nil {
+			return err
+		}
+
+		err = clearScreen()
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Orderbook streamed for %s %s:\n\n",
```

Comment:
No need for the `:` or the `\n\n` since you clean the screen for every result screened. Cool POF to show that it can receive multiple results though!